File: NLIP.iShare.Abstractions/Query.cs
namespace NLIP.iShare.Abstractions
{
    public class Query
    {
        public int? Page { get; set; }
        public int? PageSize { get; set; }
        public string SortBy { get; set; }
        public string Filter { get; set; }
        public SortOrder SortOrder { get; set; }
    }
}

File: NLIP.iShare.AuthorizationRegistry.SPA/common/generic/models/Query.ts
export class Query {
  sortBy: string;
  sortOrder: string;
  filter: string;
  pageSize: number;
  page: number;
  total: number;

  constructor(sortBy: string) {
    this.sortBy = sortBy;
    this.sortOrder = 'asc';
    this.filter = '';
    this.pageSize = 10;
    this.page = 1;
    this.total = 0;
  }
}
